src/Sarif/PerLanguageOption.cs

```diff
@@ -48,7 +48,7 @@ public PerLanguageOption(string feature, string name, Func<T> defaultValue, stri
 
             if (string.IsNullOrWhiteSpace(name))
             {
-                throw new ArgumentException(nameof(name));
+                throw new ArgumentNullException(nameof(name));
```

Comment:
This is one of two possible approaches. I'll let @michaelcfanning decide which he prefers.

This FxCop warning alerts you to a common mistake that people make when constructing an `ArgumentException`. On the web page [https://msdn.microsoft.com/en-us/library/ms182347.aspx](https://msdn.microsoft.com/en-us/library/ms182347.aspx), the Rule Description section provides the details. The problem is that in the `ArgumentException` constructor overloads, the argument is a message string, not the name of the parameter.

So you have two choices:

1. Provide a message string, for example:

```
throw new ArgumentException("You must provide a non-empty string", nameof(name));
```

(except that you would use a resource string instead of a literal string for the `message` argument).

2. Use a more specific exception type, as you have done.

The only problem with #2 -- if it is a problem -- is that if `name` is a string consisting of only white space, say `"   "`, then you will get an `ArgumentNullException` even though the argument is not technically null.

Full disclosure, I usually write the code like this:

```
if (name == null)
{
    throw new ArgumentNullException(nameof(name));
}

if (name.IsNullOrWhiteSpace(name))
{
    throw new ArgumentException(Resources.ErrorStringMustNotBeEmpty, nameof(name));
}
```

... but I don't get too worked up if other people don't do that.